### PR TITLE
Run3-sim96 Rearrange the files in SimG4CMS/Forward and collaspse some of the .h and .cc files

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Rearrange the files in SimG4CMS/Forward and collaspse some of the .h and .cc files

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 1]
```diff
@@ -29,21 +43,190 @@
...
+  int detLevels(const G4VTouchable*) const;
+  G4String detName(const G4VTouchable*, int, int) const;
+  void detectorLevel(const G4VTouchable*, int&, int*, G4String*) const;
+
+  double rinCalo, zinCalo;
+  int lastTrackID;
+  int verbosity;
+
+  // SumEnerDeposit - all deposited energy on all steps ;  SumStepl - length in steel !!!
+  G4double SumEnerDeposit, SumStepl, SumStepc;
```
Comment (perrotta) [inline SimG4CMS/Forward/plugins/BscTest.cc:204]: Class member's names shouls start with lowercase initial

File: SimG4CMS/Forward/plugins/DoCastorAnalysis.cc [hunk 2]
```diff
@@ -9,24 +9,104 @@
...
+
+  TFile *CastorOutputEventFile;
+  TTree *CastorTree;
+
+  int eventIndex;
+
+  std::vector<double> simhit_x, simhit_y, simhit_z;
+  std::vector<double> simhit_eta, simhit_phi, simhit_energy;
+  std::vector<int> simhit_sector, simhit_module;
+  //std::vector<double> simhit_time;
```
Comment (perrotta) [inline SimG4CMS/Forward/plugins/DoCastorAnalysis.cc:99]: Should `simhit_time` and `*psimhit_time` remain here (for possible future usages)?
If not, better to remove these commented out definitions, and the corresponding commented out lines in the implementation part.

**Review by perrotta — COMMENTED** [review]
@bsunanda , thank you for these cleanings and rearrangements.
Since you are cleaning, here follow a few suggestion for further cleanings of the touched files

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 3]
```diff
@@ -29,21 +43,190 @@
...
+#include "TNamed.h"
 
 //#define EDM_ML_DEBUG
 
+class BscAnalysisHistManager : public TNamed {
+public:
+  BscAnalysisHistManager(const TString& managername);
+  ~BscAnalysisHistManager() override;
+
+  TH1F* GetHisto(Int_t Number);
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/BscTest.cc]: Taken care of

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 4]
```diff
@@ -29,21 +43,190 @@
...
+                public Observer<const EndOfRun*>,
+                public Observer<const BeginOfEvent*>,
+                public Observer<const BeginOfTrack*>,
+                public Observer<const G4Step*>,
+                public Observer<const EndOfTrack*>,
+                public Observer<const EndOfEvent*> {
+public:
+  BscTest(const edm::ParameterSet& p);
+  ~BscTest() override;
+  //MyActions();
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/BscTest.cc]: Taken care of

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 5]
```diff
@@ -29,21 +43,190 @@
...
+
+private:
+  //UHB_Analysis* UserNtuples;
+  BscNumberingScheme* theBscNumberingScheme;
+
+  int iev;
+  int itrk;
+  G4double entot0, tracklength0;
+
+private:
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/BscTest.cc]: Taken care of

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 6]
```diff
@@ -29,21 +43,190 @@
...
+  TNtuple* bsceventntuple;
+  TFile bscOutputFile;
+  int whichevent;
+
+  BscAnalysisHistManager* TheHistManager;  //Histogram Manager of the analysis
+  std::string fDataLabel;                  // Data type label
+  std::string fOutputFile;                 // The output file name
+  std::string fRecreateFile;               // Recreate the file flag, default="RECREATE"
+
+  //  //  //  //  //  //  TObjString fHistType;
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/BscTest.cc]: Taken care of

File: SimG4CMS/Forward/plugins/SimG4FluxProducer.cc [hunk 7]
```diff
@@ -10,12 +23,49 @@
...
+  // observer classes
+  void update(const BeginOfRun *run) override;
+  void update(const BeginOfEvent *evt) override;
+  void update(const G4Step *step) override;
+
+  void endOfEvent(ParticleFlux &pflx, unsigned int k);
+  G4VPhysicalVolume *getTopPV();
+  std::map<G4LogicalVolume *, std::pair<unsigned int, std::string>>::iterator findLV(G4LogicalVolume *plv);
+
+private:
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/SimG4FluxProducer.cc]: Taken care of

File: SimG4CMS/Forward/plugins/ZdcTestAnalysis.cc [hunk 8]
```diff
@@ -1,15 +1,102 @@
...
+private:
+  // observer classes
+  void update(const BeginOfJob* run) override;
+  void update(const BeginOfRun* run) override;
+  void update(const EndOfRun* run) override;
+  void update(const BeginOfEvent* evt) override;
+  void update(const EndOfEvent* evt) override;
+  void update(const G4Step* step) override;
+
+private:
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/ZdcTestAnalysis.cc]: Taken care of

[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/ZdcTestAnalysis.cc]: Taken care of

File: SimG4CMS/Forward/plugins/DoCastorAnalysis.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/DoCastorAnalysis.cc]: Taken care of

File: SimG4CMS/Forward/plugins/TotemTestGem.cc [hunk 9]
```diff
@@ -11,26 +11,61 @@
...
+protected:
+  // observer classes
+  void update(const BeginOfEvent* evt) override;
+  void update(const EndOfEvent* evt) override;
+
+private:
+  void clear();
+  void fillEvent(TotemTestHistoClass&);
+
+private:
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/TotemTestGem.cc]: Taken care of

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Forward/plugins/BscTest.cc [hunk 10]
```diff
@@ -29,21 +43,190 @@
...
+  TH1F* GetHisto(Int_t Number);
+  TH1F* GetHisto(const TObjString& histname);
+
+  TH2F* GetHisto2(Int_t Number);
+  TH2F* GetHisto2(const TObjString& histname);
+
+  void WriteToFile(const TString& fOutputFile, const TString& fRecreateFile);
+
+private:
+  void BookHistos();
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/Forward/plugins/BscTest.cc]: resolved

**Comment by bsunanda** [comment]
this failure is spurious. this is happening too often

**Comment by bsunanda** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
@bsunanda thank you for fixing.
Yes, don't mind about those Relval-Input errors, they'll likely show up anyhow even if you retry testing.
If @cms-sw/simulation-l2 has nothing in contrary now, this can get merged

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Comment by qliphy** [comment]
merge

**Comment by qliphy** [comment]
merge

**Pull request merged**